opencog/cython/opencog/Cast.cc

```diff
@@ -8,3 +8,4 @@ using namespace opencog;
 Handle atom_from_the_void(long p) { return *((Handle*) p); }
 long void_from_candle(const Handle& hp) { return (long) (&hp); }
 long void_from_cptr(Handle* hp) { return (long) (hp); }
+long avoid_from_cptr(Handle* hp) { return (long) (hp->operator->()); }
```

Comment:
I think the answer is that Cython cannot work directly with C++ objects; it can only store pointers. Since handles are C++ objects, it can only store pointers to handles.

Next, cython cannot work directly with C++ pointers either, so everything must be cast into long integers. So pointers get converted to long integers, and back.  Its not type-safe, but I don't know of a better way. -- it would take a lot of work to design some better way. (which might not be better in the end)